docs/authoring.md

```diff
@@ -259,3 +225,85 @@ There are some template parameters that have a special meaning:
 |Name|Description|
 |---|---|
 |`server`| It is used to let the template know which server from the AsyncAPI specification file you want to use. In some cases, this may be required. For instance, when generating code that connects to a specific server. Use this parameter in case your template relies on users' information about what server from the specification file they want to use during generation. You also need this parameter if you want to use `server.protocol` notation within `conditionalFiles` configuration option. Once you decide to specify this parameter for your template, it is recommended you make it a mandatory parameter otherwise feature like `conditionalFiles` is not going to work if your users do not use this parameter obligatory.
+
+
+## React
+[React](https://reactjs.org) is the newest render engine which we strongly suggest you should use for any new templates. It is only the default render engine to enable backward compatibility.
+* It enables the possiblity of [debugging](#debugging-react-template) your template (this is not possible with Nunjucks).
+* It provides better stack traces once something is not working.
+* Provides better support for separating code into more manageable chunks.
+* The readability of the template are much better then Nunjucks syntax.
+* Better tool support for development.
+
+When writing react templates you decide whether to use CommonJS, ES5 or ES6 modules since everything is bundles together before the rendering process takes over. We use our own React renderer which can be found in the [Generator React SDK](https://github.com/asyncapi/generator-react-sdk). 
+There you can find information about how the renderer works or how we transpile your template files.
+
+Your React template always require 2 dependencies `react` and `@asyncapi/generator-react-sdk`. The `react` dependency is pretty self explanitory, it allows us to render the template. `@asyncapi/generator-react-sdk` is required since react files does not export default function where the root component is not of type `File` will not be rendered, furthermore it provide some support components to easy your development. 
```

Comment:
```md
...The `react` dependency is pretty self explanitory, it allows us to render the template...
```

The react is required because before transpilation you have `<Component {...props} >` but after transpilation you have `React.createElement(Component, props)` so without `import React from 'react'` then your code is broken (React will be undefined). It's a magic but it's necessary :) Also react doesn't render our template - it does by our custom renderer in `@asyncapi/generator-react-sdk` package by exported `render` method.

I would rather to change whole paragraph to something like:

```md
Your React template always require 2 dependencies `react` and `@asyncapi/generator-react-sdk`. The `react` dependency is required to declare React components inside template. `@asyncapi/generator-react-sdk` is required to import and use as a root component of your template the `File` component - it provides required metadata and without it your template won't be rendered. Furthermore it provides some common components to make your development easier, like `Text` or `Indent`
```

Reply:
Yea you are right, abstracted away from that a bit. Try see the updated readme 😄